inst/include/vctrs.h

```diff
@@ -5,6 +5,8 @@
 #include <R_ext/Rdynload.h>
 #include <stdbool.h>
 
+// Now named `obj_is_vector()` in vctrs, but we retain this for backwards
```

Comment:
I think we could add an alias so that we can start using it in dplyr, and eventually make the old name defunct.

Maybe move the old name to a deprecated section with the date of deprecation?

By the way I think that's an advantage of standalone files: We can completely remove deprecated symbols from the header file and ensure that no new dependency will use these. But I guess a deprecated section is a sufficiently strong signal.

Reply:
https://github.com/r-lib/vctrs/pull/1786/commits/8c0d087fe0dd4fa9e650492971b9e4ab0af9befc